tests/functional_tests/blockchain.py

```diff
@@ -0,0 +1,104 @@
+# permitted provided that the following conditions are met:
+# 
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+# 
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+# 
+# 3. Neither the name of the copyright holder nor the names of its contributors may be
+#    used to endorse or promote products derived from this software without specific
+#    prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Test blockchain RPC calls
+
+Test the following RPCs:
+    - get_info
+    - generateblocks
+    - [TODO: many tests still need to be written]
+
+"""
+
+from test_framework.daemon import Daemon
+from test_framework.wallet import Wallet
+
+class BlockchainTest():
+    def run_test(self):
+        self._test_get_info()
+        self._test_generateblocks(5)
+
+    def _test_get_info(self):
+        print('Test get_info')
+
+        keys = [
+                'alt_blocks_count', 
+                'block_size_limit', 
+                'block_size_median', 
+                'bootstrap_daemon_address', 
+                'cumulative_difficulty', 
+                'difficulty', 
+                'free_space', 
+                'grey_peerlist_size', 
+                'height', 
+                'height_without_bootstrap', 
+                'incoming_connections_count', 
+                'mainnet', 
+                'offline', 
+                'outgoing_connections_count', 
+                'rpc_connections_count', 
+                'stagenet',
+                'start_time', 
+                'status', 
+                'target', 
+                'target_height', 
+                'testnet',
+                'top_block_hash', 
+                'tx_count', 
+                'tx_pool_size', 
+                'untrusted', 
+                'was_bootstrap_ever_used', 
+                'white_peerlist_size'
+                ]
+
+        daemon = Daemon()
+        res = daemon.get_info()
+
+        assert sorted(res.keys()) == sorted(keys)
```

Comment:
This means every time someone adds something, this has to be updated too, for no other reason that there is a list above. That sounds a bit like busy work maybe. How about just check all keys you are testing (free space and height here) are present instead ?

Reply:
Indeed, adjusted to check space/height individually. Can be abstracted in the future to e.g. always test for existence and a predicate